### PR TITLE
Fix unjustified warning about casing for directory entries

### PR DESCRIPTION
**Pull request by mthmulders**
Follow-up for #128, where we omitted the case that an archive entry could be a directory. In that case, its name will end with `/`, but the canonical target path will not have the `/` suffix. This would be seen as a difference in casing, but that is not true.

### CODE REVIEW

File: src/test/java/org/codehaus/plexus/archiver/AbstractUnArchiverTest.java
Comment (michael-o) [inline src/test/java/org/codehaus/plexus/archiver/AbstractUnArchiverTest.java:195]: This argument is redundant.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: src/test/java/org/codehaus/plexus/archiver/AbstractUnArchiverTest.java
Comment (mthmulders) [reply src/test/java/org/codehaus/plexus/archiver/AbstractUnArchiverTest.java]: It is. Resolved.

**Review by michael-o — APPROVED** [review]

**Comment by mthmulders** [comment]
Thanks @michael-o for the review. I don't have rights to merge this PR, I'd be grateful if someone who does have the rights can merge it for me.

**Comment by mthmulders** [comment]
@michael-o I see a lot of Maven components already updating Plexus Archiver, with this little issue still present. Shouldn't we try to merge this in, too, given the false warnings that'll occur otherwise?

**Comment by plamentotev** [comment]
@mthmulders would you please squash the commits into one and I'll merge it.

**Comment by mthmulders** [comment]
> @mthmulders would you please squash the commits into one and I'll merge it.

Squashed & force-pushed. Thanks in advance :-)

**Comment by plamentotev** [comment]
The build is failing after the merge. I'll check it later today

**Comment by mthmulders** [comment]
> Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project plexus-archiver: Execution default-compile of goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile failed: Plugin org.apache.maven.plugins:maven-compiler-plugin:3.8.1 or one of its dependencies could not be resolved: Could not transfer artifact org.apache.maven.shared:maven-shared-incremental:jar:1.1 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-incremental/1.1/maven-shared-incremental-1.1.jar: Connection reset -> [Help 1]

My guess is that's unrelated to the code change, but feel free to investigate further and let me know if you draw a different conclusion.

**Comment by michael-o** [comment]
> 
> 
> > Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project plexus-archiver: Execution default-compile of goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile failed: Plugin org.apache.maven.plugins:maven-compiler-plugin:3.8.1 or one of its dependencies could not be resolved: Could not transfer artifact org.apache.maven.shared:maven-shared-incremental:jar:1.1 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-incremental/1.1/maven-shared-incremental-1.1.jar: Connection reset -> [Help 1]
> 
> My guess is that's unrelated to the code change, but feel free to investigate further and let me know if you draw a different conclusion.

Correct, is an ephemeral network issue. Restart or check your internet connection.

**Comment by plamentotev** [comment]
@mthmulders I've released a new version with this fix. Thanks for contributing.

**Comment by mthmulders** [comment]
Thank you for considering this contribution and for releasing a new version so quickly.

**Pull request merged**
